spec/facets/DatasetVersionDatasetFacet.json

```diff
@@ -0,0 +1,33 @@
+{
+  "$schema" : "https://json-schema.org/draft/2020-12/schema",
+  "$id" : "https://openlineage.io/spec/facets/1-0-0/DatasetVersionDatasetFacet.json",
+  "$defs" : {
+    "DatasetVersionDatasetFacet": {
+      "allOf": [
+        {
+          "$ref": "https://openlineage.io/spec/1-0-2/OpenLineage.json#/$defs/DatasetFacet"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "datasetVersion": {
+              "description": "The version of the dataset.",
+              "type": "string"
+            }
+          },
+          "additionalProperties": true,
+          "required": [
+            "datasetVersion"
+          ]
+        }
+      ],
+      "type": "object"
+    }
+  },
+  "type" : "object",
+  "properties" : {
+    "version" : {
+      "$ref" : "#/$defs/DatasetVersionDatasetFacet"
```

Comment:
From my comment [above](https://github.com/OpenLineage/OpenLineage/pull/580/files#r819366769), I would reverse the facet property naming to go from specific to more general naming strategy which is a bit more intuitive when chaining the method calls:

```java
dataset.getFacets().getDatasetVersion().getVersion()
```

Reply:
On one hand, from the generated Java client code perspective, this looks better. 
However, there used to be some discussion about that in the past:
https://github.com/OpenLineage/OpenLineage/issues/35#issuecomment-1004741548
that resulted in `DatasetVersionDatasetFacet` over `VersionDatasetFacet`
which looks better from JSON spec files point of view as one instantly gets to know what `version` is the facet about. 

I can't see any significant benefits of one approach over the another, and would leave it like that.